kai/evaluation.py

```diff
@@ -0,0 +1,145 @@
+# Make our own test data
+
+# provide full kai config (params, templates, which model i'm using, etc...)
+
+# provider = "IBMGranite"
+# args = { model_id = "ibm/granite-13b-chat-v2" }
+
+# Prompt builder stuff inside examples/prompt_engineering.ipynb
+
+# provide multiple sets
+# run it through the suite and say "hey yo this is worse or better"
+
+import argparse
+import os
+import tomllib
+from dataclasses import dataclass
+
+import yaml
+
+from kai.model_provider import ModelProvider
+
+DEFAULT_CONFIGURATION = {
+    "models": {
+        "provider": "IBMOpenSource",
+        "args": {"model_id": "ibm-mistralai/mixtral-8x7b-instruct-v01-q"},
+    }
+}
+
+
+@dataclass
+class MockIncidentStore:
+    expected: str
+    original_file: str
+    incident_store: dict
+
+
+@dataclass
+class BenchmarkExample:
+    expected: str
+    original_file: str
+    incident_store: MockIncidentStore
+
+
+# Application:
+#     Ruleset:
+#         Violation:
+#             Incidents:
+#                 uri:
+#                 snip:
+#                 line:
+#                 variables:
+
+
+def load_benchmarks(
+    examples_path=os.path.join(os.path.dirname(__file__), "data", "benchmarks")
```

Comment:
Github already called it out, but we shouldn't be doing these function calls in the function arguments